async/index.d.ts

```diff
@@ -6,17 +6,16 @@
 interface Dictionary<T> { [key: string]: T; }
 
 interface ErrorCallback<T> { (err?: T): void; }
-interface AsyncWaterfallCallback<E> { (err: E, ...args: any[]): void; }
-interface AsyncBooleanResultCallback<E> { (err: E, truthValue: boolean): void; }
-interface AsyncResultCallback<T, E> { (err: E, result: T): void; }
-interface AsyncResultArrayCallback<T, E> { (err: E, results: T[]): void; }
-interface AsyncResultObjectCallback<T, E> { (err: E, results: Dictionary<T>): void; }
+interface AsyncBooleanResultCallback<E> { (err?: E, truthValue?: boolean): void; }
+interface AsyncResultCallback<T, E> { (err?: E, result?: T): void; }
+interface AsyncResultArrayCallback<T, E> { (err?: E, results?: (T | undefined)[]): void; }
+interface AsyncResultObjectCallback<T, E> { (err: E | undefined, results: Dictionary<T>): void; }
 
-interface AsyncFunction<T, E> { (callback: (err?: E, result?: T) => void): void; }
+interface AsyncFunction<T, E> { (callback: (err?: E, result?: T) => void): any; }
```

Comment:
I thought that a function declared as returning `void` actually had to return `void` (since it works that way with variables). 
Thanks for explaining that, I'll fix it.